src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -0,0 +1,904 @@
+        }
+
+        /// <summary>
+        /// Adds the specified number of months to the value of this instance.
+        /// </summary>
+        /// <param name="value">A number of months. The months parameter can be negative or positive.</param>
+        /// <returns>An object whose value is the sum of the date represented by this instance and months.</returns>
+        public DateOnly AddMonths(int value) => new DateOnly(DayNumberFromDateTime(GetEquivalentDateTime().AddMonths(value)));
+
+        /// <summary>
+        /// Adds the specified number of years to the value of this instance.
+        /// </summary>
+        /// <param name="value">A number of years. The value parameter can be negative or positive.</param>
+        /// <returns>An object whose value is the sum of the date represented by this instance and the number of years represented by value.</returns>
+        public DateOnly AddYears(int value) => new DateOnly(DayNumberFromDateTime(GetEquivalentDateTime().AddYears(value)));
+
+        /// <summary>
+        /// Determines whether two specified instances of DateOnly are equal.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <returns>true if left and right represent the same date; otherwise, false.</returns>
+        public static bool operator ==(DateOnly left, DateOnly right) => left._dayNumber == right._dayNumber;
+
+        /// <summary>
+        /// Determines whether one specified DateOnly is later than another specified DateTime.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <returns>true if left is later than right; otherwise, false.</returns>
+        public static bool operator >(DateOnly left, DateOnly right) => left._dayNumber > right._dayNumber;
+
+        /// <summary>
+        /// Determines whether one specified DateOnly represents a date that is the same as or later than another specified DateOnly.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <returns>true if left is the same as or later than right; otherwise, false.</returns>
+        public static bool operator >=(DateOnly left, DateOnly right) => left._dayNumber >= right._dayNumber;
+
+        /// <summary>
+        /// Determines whether two specified instances of DateOnly are not equal.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <returns>true if left and right do not represent the same date; otherwise, false.</returns>
+        public static bool operator !=(DateOnly left, DateOnly right) => left._dayNumber != right._dayNumber;
```

Comment:
Nit: move this up to be just after the `operator ==` ?

Reply:
I'll do that.